lib/components/jpegdepay/jpegdepay.js

```diff
@@ -0,0 +1,103 @@
+const { Rtp } = require('../../utils/protocols')
+const {
+  makeImageHeader,
+  makeQuantHeader,
+  makeFrameHeader,
+  makeHuffmanHeader,
+  makeScanHeader
+} = require('./headers')
+
+/**  Each packet contains a special JPEG header which immediately follows
+   the RTP header.  The first 8 bytes of this header, called the "main
+   JPEG header", are as follows:
+
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   | Type-specific |              Fragment Offset                  |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |      Type     |       Q       |     Width     |     Height    |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+ */
+
+/** Restart Marker header: when using types 64-127
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |       Restart Interval        |F|L|       Restart Count       |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+
+/** Quantization Table header: when using Q values 128-255
+    0                   1                   2                   3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |      MBZ      |   Precision   |             Length            |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                    Quantization Table Data                    |
+   |                              ...                              |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+ */
+
+module.exports = function jpegDepayFactory (width, height) {
+  const IMAGE_HEADER = makeImageHeader()
+  const HUFFMAN_HEADER = makeHuffmanHeader()
+  const SCAN_HEADER = makeScanHeader()
+
+  return function jpegDepay (packets) {
+    let metadata
+    const fragments = packets.map((packet) => {
+      let payload = Rtp.payload(packet)
+
+      // Parse and extract JPEG header.
+      const typeSpecific = payload.readUInt8(0)
+      const fragmentOffset = payload.readUInt8(1) << 16 | payload.readUInt8(2) << 8 | payload.readUInt8(3)
+      const type = payload.readUInt8(4)
+      const Q = payload.readUInt8(5)
+      // const width = payload.readUInt8(6) * 8
```

Comment:
Remove?

Reply:
I wanted to leave the comments there to show how width and height need to be defined. I should add a comment that the width and height are taken from the session description protocol instead, in order to provide support for sizes that are beyond the JPEG-RTP maximum width/height.